From ad2e96d559afe6dc43c040d213f1bc7cf040297c Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Fri, 24 Mar 2023 17:32:53 +0100
Subject: [PATCH 17/25] debugedit: skip .debug_types tests if compiler doesn't
 generate section

Some compilers don't generate a .debug_types section even when compiling
with -fdebug-types-section. Skip testing the .debug_types tests in that
case.

Signed-off-by: Mark Wielaard <mark@klomp.org>
---
 tests/debugedit.at | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/debugedit.at b/tests/debugedit.at
index 725e68e..c0fb343 100644
--- a/tests/debugedit.at
+++ b/tests/debugedit.at
@@ -364,6 +364,7 @@ AT_CLEANUP
 AT_SETUP([debugedit .debug_types objects])
 AT_KEYWORDS([debugtypes] [debugedit])
 DEBUGEDIT_SETUP([-fdebug-types-section -gdwarf-4])
+AT_SKIP_IF([! $READELF -S ./foo.o ./subdir_bar/bar.o ./baz.o | grep -F -q '.debug_types'])
 
 AT_DATA([expout],
 [st1
@@ -401,6 +402,7 @@ AT_CLEANUP
 AT_SETUP([debugedit .debug_types partial])
 AT_KEYWORDS([debugtypes] [debugedit])
 DEBUGEDIT_SETUP([-fdebug-types-section -gdwarf-4])
+AT_SKIP_IF([! $READELF -S ./foobarbaz.part.o | grep -F -q '.debug_types'])
 
 AT_DATA([expout],
 [st1
@@ -430,6 +432,7 @@ AT_CLEANUP
 AT_SETUP([debugedit .debug_types exe])
 AT_KEYWORDS([debugtypes] [debugedit])
 DEBUGEDIT_SETUP([-fdebug-types-section -gdwarf-4])
+AT_SKIP_IF([! $READELF -S ./foobarbaz.exe | grep -F -q '.debug_types'])
 
 AT_DATA([expout],
 [st1
-- 
2.43.0

